### PR TITLE
REST reactive: use ClientLogger bean, if present

### PR DESCRIPTION
**Pull request by mickroll**
With `org.jboss.resteasy.reactive.client.api.ClientLogger`, there is already an api to implement a custom REST reactive client logger. For now, the only way to use it, is to set it while building `org.jboss.resteasy.reactive.client.impl.ClientBuilderImpl` manually. All other REST reactive clients fall back to using `org.jboss.resteasy.reactive.client.logging.DefaultClientLogger`.

This PR adds bean pickup logic: if a CDI bean for `ClientLogger` is found, it is used. Otherwise, use `DefaultClientLogger`.

As Arc is not a dependency in the `org.jboss.resteasy.reactive.client` context (and shouldn't be if i understand correctly), i added the pickup logic to `io.quarkus.rest.client.reactive.runtime.RestClientBuilderImpl`, which in turn uses `ClientBuilderImpl`.
But there is another place where `ClientBuilderImpl` is used: `io.quarkus.keycloak.admin.client.reactive.runtime.ResteasyReactiveClientProvider`. So i added the pickup logic there as well.

The solution was discussed here: https://github.com/quarkusio/quarkus/discussions/33332

cc @michalvavrik, @geoand

### CODE REVIEW

**Comment by geoand** [comment]
cc @Sgitario

File: extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/RestClientBuilderImpl.java [hunk 1]
```diff
@@ -65,6 +66,13 @@ public class RestClientBuilderImpl implements RestClientBuilder {
     private String proxyPassword;
     private String nonProxyHosts;
 
+    public RestClientBuilderImpl() {
+        InstanceHandle<ClientLogger> clientLogger = Arc.container().instance(ClientLogger.class);
```
Comment (Sgitario) [inline extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/RestClientBuilderImpl.java:70]: This should be done in the `build` method.

File: docs/src/main/asciidoc/rest-client-reactive.adoc [hunk 2]
```diff
@@ -1364,6 +1364,8 @@ quarkus.rest-client.logging.body-limit=50
 quarkus.log.category."org.jboss.resteasy.reactive.client.logging".level=DEBUG
 ----
 
+REST Client Reactive uses a default `ClientLogger` implementation. You can change it by providing a custom `ClientLogger` instance via CDI or when programmatically creating your client.
```
Comment (Sgitario) [inline docs/src/main/asciidoc/rest-client-reactive.adoc:1367]: I will add this sentence as a TIP. Also, fix the reported issue by Linting with Vale.

**Review by Sgitario — CHANGES_REQUESTED** [review]
I would add a simple test case that ensures this works when:
- Injecting the REST Client via `@RestClient`
- Programmatically using QuarkusRestClientBuilderImpl
- Programmatically using RestClientBuilderImpl

File: docs/src/main/asciidoc/rest-client-reactive.adoc (hunk 2 above)
Comment (mickroll) [reply docs/src/main/asciidoc/rest-client-reactive.adoc:1367]: >I will add this sentence as a TIP.
Sorry to ask, but what do you mean by that?

I will fix the linting issue.

File: extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/RestClientBuilderImpl.java (hunk 1 above)
Comment (michalvavrik) [reply extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/RestClientBuilderImpl.java:70]: If this is done in the `build` method, then you have no chance to override this by setting `clientLogger` on builder. As long as there is bean, you can't  set your own logger with builder.

Comment (michalvavrik) [reply extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/RestClientBuilderImpl.java:70]: No, you are right, you don't have there access to logger anyway.

**Comment by Sgitario** [comment]
> > I would add a simple test case that ensures this works when:
> > * Injecting the REST Client via `@RestClient`
> > * Programmatically using QuarkusRestClientBuilderImpl
> > * Programmatically using RestClientBuilderImpl
> 
> @Sgitario I added both a test using `@RestClient` and a test using `QuarkusRestClientBuilderImpl`. The latter internally delegates to `RestClientBuilderImpl` so i did not write an extra test for that. I tried to keep close to how `ResponseExceptionMapper` is tested.

Please, be sure you build the modules you have modified using the Maven command before pushing the commit to avoid formatting issues. For example, running `mvn clean install -Dquickly` would do it. 

Also, the linting issue with the documentation still stands.

**Comment by mickroll** [comment]
Sorry for the push-mess in this PR.
I have finally managed to setup a local dev environment.

**Review by Sgitario — APPROVED** [review]

**Comment by Sgitario** [comment]
@michalvavrik do you want to take another look before merging this one?

**Comment by michalvavrik** [comment]
thanks @Sgitario , `keycloak-admin-client-reactive` part LGTM

**Comment by mickroll** [comment]
Thanks for your patience, @Sgitario and @michalvavrik

**Comment by Sgitario** [comment]
> Thanks for your patience, @Sgitario and @michalvavrik

Thanks to you for your work!

**Pull request merged**